### PR TITLE
Add multiple python versions to CI tests

### PR DESCRIPTION
**Pull request by shaunagm**
We had a pair programming session where we attempted to address #790 within CircleCI, that was unsuccessful so we switched to Github Actions.

- [x] get tests working in github actions
- [x] add caching
- [x] move Windows error to new issue #860 
  - see [this windows error](https://github.com/move-coop/parsons/actions/runs/5718559743/job/15494634900?pr=858)
- [x] check usage requirements to make sure we won't run out of minutes/storage
   - a single run of the tests (both Linux and Mac) is about 5 minutes, and we seem to have 2000 free minutes total, so we have about 400 runs per month or an average of 12-13 test runs per day average. that may not be enough but it may be fine? Buying additional minute seems to cost about $10 per thousand minutes so it's not a huge deal if we overrun occasionally
- [x] check that the docs build stuff is still working on circle CI

### CODE REVIEW

File: .github/workflows/test-linux-windows.yml [hunk 1]
```diff
@@ -0,0 +1,50 @@
+name: tests
+
+on:
+  pull_request:
```
Comment (gcollazo) [inline .github/workflows/test-linux-windows.yml:4]: Consider adding "on push" so tests run once more once PRs are merged into main

```yml
on:
  push:
    branches: ["main"]
```

File: .github/workflows/tests-mac.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
+name: tests for mac
+# test mac on single python version as mac tests use 10x minutes/storage
+
+on:
```
Comment (gcollazo) [inline .github/workflows/tests-mac.yml:4]: Consider adding "on push"

**Review by gcollazo — COMMENTED** [review]
Thank you for moving this to GH actions 👍

A few comments:

- Tests not running after PR is merged into `main`. Consider adding "on push".
- For Mac it's only running on Python 3.8. Does that satisfy the requirements?
- Not testing withPython 3.7. Consider updating README, docs and setup.py to deprecate support of 3.7.

**Comment by shaunagm** [comment]
Thanks for the feedback @gcollazo! I updated to address suggestions 1 & 3. Regarding question 2 - we have a set number of minutes we can use per month, and tests on Mac are given a 10x multiplier, so something that takes 3 minutes costs 30 minutes (as opposed to 3 minutes on Linux, and 6 minutes on Mac). I don't think we have enough free minutes to actually run that many Mac tests, so I'm hoping that just running on one version of Python will catch any Mac-specific errors.

**Pull request merged**